src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1684,93 +1609,40 @@ public override Task<T> GetFieldValueAsync<T>(int ordinal, CancellationToken can
     /// <typeparam name="T">Synchronously gets the value of the specified column as a type.</typeparam>
     /// <param name="ordinal">The column to be retrieved.</param>
     /// <returns>The column to be retrieved.</returns>
-    public override T GetFieldValue<T>(int ordinal)
-    {
-        if (typeof(T) == typeof(Stream))
-            return (T)(object)GetStream(ordinal);
+    public override T GetFieldValue<T>(int ordinal) => GetFieldValueCore<T>(ordinal);
 
-        if (typeof(T) == typeof(TextReader))
-            return (T)(object)GetTextReader(ordinal);
-
-        if (_isSequential)
-            return GetFieldValueSequential<T>(ordinal, false).GetAwaiter().GetResult();
-
-        // In non-sequential, we know that the column is already buffered - no I/O will take place
-
-        var field = CheckRowAndGetField(ordinal);
-        SeekToColumnNonSequential(ordinal);
-
-        if (ColumnLen == -1)
-        {
-            // When T is a Nullable<T> (and only in that case), we support returning null
-            if (NullableHandler<T>.Exists)
-                return default!;
-
-            if (typeof(T) == typeof(object))
-                return (T)(object)DBNull.Value;
-
-            ThrowHelper.ThrowInvalidCastException_NoValue(field);
-        }
-
-        // We don't handle exceptions or update PosInColumn
-        // As with non-sequential reads we always just move to the start/end of the column
-        return NullableHandler<T>.Exists
-            ? NullableHandler<T>.Read(field.Handler, Buffer, ColumnLen, field)
-            : typeof(T) == typeof(object)
-                ? (T)field.Handler.ReadAsObject(Buffer, ColumnLen, field)
-                : field.Handler.Read<T>(Buffer, ColumnLen, field);
-    }
-
-    async ValueTask<T> GetFieldValueSequential<T>(int column, bool async, CancellationToken cancellationToken = default)
+    T GetFieldValueCore<T>(int ordinal)
     {
-        using var registration = Connector.StartNestedCancellableOperation(cancellationToken, attemptPgCancellation: false);
-
-        var field = CheckRowAndGetField(column);
-        await SeekToColumnSequential(column, async, CancellationToken.None);
-        CheckColumnStart();
-
-        if (ColumnLen == -1)
-        {
-            // When T is a Nullable<T> (and only in that case), we support returning null
-            if (NullableHandler<T>.Exists)
-                return default!;
+        // The only statically mapped converter, it always exists.
+        if (typeof(T) == typeof(Stream))
```

Comment:
I hate these Stream/TextReader checks, how about we remove them (very minor breaking changing IMHO)? I know these checks are compiled out for value types, but still.

Reply:
It would certainly help move perf a few ns :p